### DISCUSSION TITLE
Bots should keep doing CPR when I switch off them

### DISCUSSION THREAD

**Discussion by pantherpusher** [FakeFishGames/Barotrauma#11802] [Ideas and suggestions]
As is well known, in singleplayer you play as some kind of puppet master that can control any crew member at any time, possessing them demonically, likely a curse related to their ultimate fate as being part of the crew. But I haven't finished the campaign before so no spoilers. I've noticed that while testing something else, if I have disabled crew AI and have a crewmate performing CPR and switch away from them to another crewmate, the first crewmate (who is now released from my direct control) continues performing CPR! This made me very excited to see! I thought to myself "This is great, as this means that any crewmember with high enough medical to not cause more harm than good doing CPR, as though they had basic first aid training, can keep a critical crewmate stable long enough for me to pilot a medic over to resuscitate them!" But alas when I reenabled crew ai, the bot immediately stopped their lifesaving efforts and stood up, doing nothing at all and leaving the poor man to expire of hypoxia at his feet. To say I was disappointed would be an understatement. Thereby I want to suggest that the change be made so that a bot that has begun CPR will not stop doing CPR unless:
1. They can no longer do CPR (The patient has died/the patient has woken up/the rescuer has been rendered unconscious).
2. The player has manually stopped them doing CPR by taking control of them again.
3. The bot doing CPR is in serious danger such as intruders, fire, flooding.
4. The bot doing CPR is given an order/told to check their priorities.

**Comment by pantherpusher** [comment]
To help make it easier to tell which bot is performing CPR I'd also suggest using the grey icon used often for showing when a bot is idle or doing something on their own volition to show a special icon to represent that they are performing CPR and on who. I believe that these changes would improve the singleplayer experience for a lot of people, even if not in ways immediately apparent. I would have put this after the list but I couldn't figure how to without it being formatted weirdly, sorry.

**Reply by Hazmat616** [reply]
What all does the rescue crew order do? Can they do CPR under it?
